### PR TITLE
feat: fully support Multi-window switch recording, etc...

### PR DESCRIPTION
**Pull request by smildlzj**
**Thanks for contributing to the Selenium IDE!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium-ide/blob/master/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

####  feat: tell renderer, navigator.webdriver = true
same behavior as other tese tools. tell the page is in test mode.

####  feat: fully support Multi-window switch recording
auto add `selectWindow` command when multi-window switch recording.

##### How to do that?
* all windows has `winHandleId` identity which get from main process.
* `winHandleId` generate by command list
* the entry window name `root`

#### support play options `delay`
image

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium-ide/blob/master/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by smildlzj** [comment]
Question: this part of code is mean anying?  

https://github.com/SeleniumHQ/selenium-ide/blob/0b1ef9e73001350436bd4220cfbf107ebecd0535/packages/selenium-ide/src/main/session/controllers/Recorder/index.ts#L71-L78

**Comment by toddtarsi** [comment]
@smildlzj - Woah! This is very impressive. I was not expecting this today. You have really dug into the codebase it seems like, wow!

**Comment by toddtarsi** [comment]
> Question: this part of code is mean anying?
> 
> https://github.com/SeleniumHQ/selenium-ide/blob/0b1ef9e73001350436bd4220cfbf107ebecd0535/packages/selenium-ide/src/main/session/controllers/Recorder/index.ts#L71-L78

@smildlzj - Sorry I didn't respond to this sooner. Yes, this part learns of new windows during recording steps. Basically, after every recorded step, we check if a new window was added. If so, the last command gets updated with the opensWindow and handle id flags. More specifically, the handle id flag is just a random-ish string which I think gets set as a variable name to the window handle or something like that.

File: packages/side-api/src/commands/recorder/recordNewCommand.ts [hunk 1]
```diff
@@ -34,14 +35,51 @@ const processSelectFrameCommands = (
...
+          i - 1,
+          [],
+        ],
+        result: [storeWindowHandle],
+      })
+      return updateStepSelection(sesh, {
+        params: [activeIndex + 1, false, false, true],
+        result: undefined,
+      })
+    } else if (item.command == 'storeWindowHandle' && target == 'root') {
```
Comment (toddtarsi) [inline packages/side-api/src/commands/recorder/recordNewCommand.ts]: @smildlzj - Super nitpick, but can these be strong equality?

File: packages/selenium-ide/src/main/session/controllers/Recorder/index.ts [hunk 2]
```diff
@@ -145,9 +201,9 @@ export default class RecorderController extends BaseController {
         playbackWindow.webContents.loadURL(`${state.project.url}/`)
         return randomUUID()
       }
-      playbackWindow.webContents.loadURL(
-        `${state.project.url}${currentCommand.target}`
-      )
+      let url = new URL(currentCommand.target as string, state.project.url)
```
Comment (toddtarsi) [inline packages/selenium-ide/src/main/session/controllers/Recorder/index.ts:204]: Wow, does this make things work in a relative and non-relative way?

File: packages/side-api/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@seleniumhq/side-api",
-  "version": "4.0.0-alpha.15",
+  "version": "4.0.0-alpha.16",
```
Comment (toddtarsi) [inline packages/side-api/package.json:3]: I can't emphasize how impressed I am with this code really. My convoluted hoisted shapes, ridiculous internal versioning, and spaghetti code was all handled flawlessly by you. This is very impressive to me. I haven't seen anyone propose code this well written without asking any questions or anything here yet.

**Review by toddtarsi — APPROVED** [review]
LGTM 👍 @smildlzj - I see nothing here to critique. This code looks very well researched and tested. If you tell me its okay to merge, I will. 😄 Nice job!!!

**Comment by smildlzj** [comment]
> LGTM 👍 @smildlzj - I see nothing here to critique. This code looks very well researched and tested. If you tell me its okay to merge, I will. 😄 Nice job!!!

Yes, it had test by myself. it work

File: packages/selenium-ide/src/main/session/controllers/Recorder/index.ts (hunk 2 above)
Comment (smildlzj) [reply packages/selenium-ide/src/main/session/controllers/Recorder/index.ts:204]: when project url is "https://xxxx.com", and open command is "https://yyyy.com".  then faill to load

**Comment by smildlzj** [comment]
> > Question: this part of code is mean anying?
> > https://github.com/SeleniumHQ/selenium-ide/blob/0b1ef9e73001350436bd4220cfbf107ebecd0535/packages/selenium-ide/src/main/session/controllers/Recorder/index.ts#L71-L78
> 
> @smildlzj - Sorry I didn't respond to this sooner. Yes, this part learns of new windows during recording steps. Basically, after every recorded step, we check if a new window was added. If so, the last command gets updated with the opensWindow and handle id flags. More specifically, the handle id flag is just a random-ish string which I think gets set as a variable name to the window handle or something like that.

is this scene?
* click to play
* click any link to open new win
* start to record on new win

**Comment by toddtarsi** [comment]
@smildlzj - You're merged! Also, yeah that seems good. Unless clicking play is also causing record. In which case, I will need to fix.

**Pull request merged**